app/src/main/java/com/dkanada/gramophone/service/MusicService.java

```diff
@@ -128,32 +115,76 @@
 
     private Playback playback;
 
-    private List<Song> playingQueue = new ArrayList<>();
-    private List<Song> originalPlayingQueue = new ArrayList<>();
-
-    private int position = -1;
-    private int nextPosition = -1;
+    public QueueManager queueManager;
 
-    private int shuffleMode;
-    private int repeatMode;
-
-    private boolean notHandledMetaChangedForCurrentTrack;
     private boolean queuesRestored;
 
     private PlayingNotification playingNotification;
     private MediaSessionCompat mediaSession;
     private PowerManager.WakeLock wakeLock;
 
-    private PlaybackHandler playerHandler;
     private Handler uiThreadHandler;
-    private ThrottledSeekHandler throttledSeekHandler;
     private QueueHandler queueHandler;
     private ProgressHandler progressHandler;
 
-    private HandlerThread playerHandlerThread;
     private HandlerThread progressHandlerThread;
     private HandlerThread queueHandlerThread;
 
+    public final QueueManager.QueueCallbacks queueCallbacks = new QueueManager.QueueCallbacks() {
```

Comment:
I prefer these implemented by the classes themselves even if it can be slightly more annoying to manage them.